zrml/prediction-markets/src/lib.rs

```diff
@@ -1379,6 +1392,8 @@ mod pallet {
                 T::Swap::set_pool_as_stale(&market.market_type, pool_id, outcome_report)?;
             }
 
+            T::LiquidityMining::distribute_market_incentives(market_id)?;
```

Comment:
`distribute_market_incentives` is marked transactional but the calling function is not, which means that a pool can still be changed to stale without distributing rewards. We should probably make the calling function transactional as well.

Reply:
Added `#[frame_support::transactional]`